flight/core/Dispatcher.php

```diff
@@ -149,62 +184,69 @@ public function filter(array $filters, array &$params, &$output): void
     /**
      * Executes a callback function.
      *
-     * @param callable|array<class-string|object, string> $callback Callback function
-     * @param array<int, mixed>          $params   Function parameters
-     *
-     * @throws Exception
+     * @param callable-string|(Closure(): mixed)|array{class-string|object, string} $callback
+     * Callback function
+     * @param array<int, mixed> $params Function parameters
      *
      * @return mixed Function results
+     * @throws Exception
      */
     public static function execute($callback, array &$params = [])
     {
-        if (\is_callable($callback)) {
-            return \is_array($callback) ?
-                self::invokeMethod($callback, $params) :
-                self::callFunction($callback, $params);
+        if (is_string($callback) && !function_exists($callback)) {
+            throw new InvalidArgumentException('Invalid callback specified.');
         }
 
-        throw new InvalidArgumentException('Invalid callback specified.');
+        if (is_array($callback)) {
+            return self::invokeMethod($callback, $params);
+        }
+
+        return self::callFunction($callback, $params);
     }
 
     /**
      * Calls a function.
      *
-     * @param callable|string $func   Name of function to call
-     * @param array<int, mixed>           $params Function parameters
+     * @param callable $func Name of function to call
+     * @param array<int, mixed> &$params Function parameters
      *
      * @return mixed Function results
      */
-    public static function callFunction($func, array &$params = [])
+    public static function callFunction(callable $func, array &$params = [])
     {
         return call_user_func_array($func, $params);
     }
 
     /**
      * Invokes a method.
      *
-     * @param mixed $func   Class method
-     * @param array<int, mixed> $params Class method parameters
+     * @param array{class-string|object, string} $func Class method
+     * @param array<int, mixed> &$params Class method parameters
      *
      * @return mixed Function results
+     * @throws TypeError For unexistent class name.
      */
-    public static function invokeMethod($func, array &$params = [])
+    public static function invokeMethod(array $func, array &$params = [])
     {
         [$class, $method] = $func;
 
-        $instance = \is_object($class);
+        if (is_string($class) && class_exists($class)) {
+            $class = new $class();
```

Comment:
Honestly this change makes me happy. This is something on my mental todo list! I was going to take it a step further where you could if you wanted to just do something like `Flight::route('/url', 'Controller->method');` but this is good for now

Reply:
Precisely for this reason I wanted to add a DI container, which is just used for dependencies of a constructor.